app/src/main/java/org/simple/clinic/searchresultsview/PatientSearchView.kt

```diff
@@ -82,9 +83,22 @@ class PatientSearchView(context: Context, attrs: AttributeSet) : RelativeLayout(
           .clicks(newPatientButton)
           .map { RegisterNewPatientClicked }
 
-  fun updateSearchResults(results: List<SearchResultsItemType<out ViewHolder>>) {
-    results.forEach { it.uiEvents = downstreamUiEvents }
-    adapter.update(results)
+  fun updateSearchResults(
+      results: PatientSearchResults,
+      currentFacility: Facility
+  ) {
+    if (results.hasNoResults) {
+      setEmptyStateVisible(true)
+      adapter.update(emptyList())
+    } else {
+      setEmptyStateVisible(false)
+      SearchResultsItemType
+          .generateListItems(results, currentFacility)
+          .let { listItems ->
+            listItems.forEach { it.uiEvents = downstreamUiEvents }
```

Comment:
Can we not pass the `downstreamUiEvents` to the adapter once, so that we don't have to set it for every single item on the list?

Reply:
We could, I'll do this.

Reply:
I took a look at this right now and I realised that this is coupled to `GroupieItemWithUiEvents` and will require extra work to fix. I am planning to extract a generic version of `AllPatientsInFacilityListAdapter` later and this will enforce a contract on how to handle these events from list items.